src/basic/procfs-util.c

```diff
@@ -201,6 +201,41 @@ int procfs_cpu_get_usage(nsec_t *ret) {
         return 0;
 }
 
+int convert_meminfo_value_to_uint64_bytes(char *word, uint64_t *ret) {
```

Comment:
i'd really prefer if this wouldn't mangle the input param. Thoe are a bit dirty, in particular in "library" code, like this here which is in src/basic/.

There's nothing wrong in strndup()ing the string before passing it to safe_atou64(). Yes, it' a memory allocation more, but it shouldn't matter, this is not called in a tight loop. And if you must you could also d a size check first, and then use strndupa(), than it would be hardly distinguishable performance-wise from the current code but vastly prettier.

Please let's not be sloppy on things like this: allow parameters to be mangled like this sooner or later means the whole codebase gets harder to work with.